### ISSUE TITLE
Build & tests fails under debian

### ISSUE THREAD

**Issue by kiorky** [grke/burp#766]
We have locally a procedure to rebuild with official debian packaging current version of burp.
- [this repo](https://github.com/corpusops/burp) builds this [PPA](https://launchpad.net/~corpusops/+archive/ubuntu/burp/+packages)

This is falling on latest master, tests arent passing anymore.

To reproduce
```sh
git clone git@github.com:corpusops/burp.git # || git pull
git reset --hard origin/issue766
docker build -t burp . -f mc_packaging/Dockerfile
```

This will eventually fail with
```
98%: Checks: 358, Failures: 4, Errors: 0
utest/client/test_extrameta.c:97:F:Core:test_extrametas:0: Assertion 'has_extrameta(path, cmd, 1, 1 )==0' failed
utest/client/protocol2/test_rabin_read.c:79:F:Core:test_rabin_read:0: Assertion 'has_xattr(sb->path.buf)==0' failed
utest/client/test_xattr.c:148:F:Core:test_xattrs:0: Assertion 'has_xattr(path)==0' failed
utest/builders/build_asfd_mock.c:78:F:Core:test_proto1_stuff:0: Assertion '!memcmp(expected->buf, wbuf->buf, wbuf->len)' failed
FAIL runner (exit status: 1)

============================================================================
Testsuite summary for Burp 2.2.12
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to https://github.com/grke/burp/issues
============================================================================
Makefile:8248: recipe for target 'test-suite.log' failed
make[3]: *** [test-suite.log] Error 1
make[3]: Leaving directory '/src'
Makefile:8354: recipe for target 'check-TESTS' failed
make[2]: *** [check-TESTS] Error 2
Makefile:8559: recipe for target 'check-am' failed
make[1]: *** [check-am] Error 2
dh_auto_test: make -j4 -O check VERBOSE=1 returned exit code 2
debian/rules:7: recipe for target 'build' failed
make: *** [build] Error 25
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
debuild: fatal error at line 1152:
dpkg-buildpackage -rfakeroot -us -uc -ui -i -b failed

```

**Comment by grke** [comment]
Hello,

Thank you for the instructions for reproducing, I followed them easily, but got slightly different results.
I have no xattr/extrameta errors.
This may be because my host supports xattrs.

I docker exec-d into the container whilst the build was running, and ran this:

```
root@c6dd50b2a689:/# mount|grep xattr
cgroup on /sys/fs/cgroup/systemd type cgroup (ro,nosuid,nodev,noexec,relatime,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd)
```
(same output when I run the command on the host)

But I do get the test_proto1_stuff error:
```
utest/builders/build_asfd_mock.c:78:F:Core:test_proto1_stuff:0: Assertion '!memcmp(expected->buf, wbuf->buf, wbuf->len)' failed
```

Suspect this is something to do with zlib versions. I will investigate that one more.

**Comment by grke** [comment]
It is to do with gzip writing different headers according to what type of OS it is running on.
I assumed that, when travis failed a couple of months ago, it was because of a different gzip version.
It's not, it is because of a different OS type. I will fix that.

**Comment by grke** [comment]
There's a patch in master now, to fix the test_proto1_stuff issue.

**Comment by kiorky** [comment]
With 61a1a1f9 (HEAD of my issue766 branch), i still have a problem  (underlying FS is ext4), but seems it's new :-).

I did not have ```user_xattr``` enabled, but even with activation, still no luck, that same new error :

```
99%: Checks: 358, Failures: 3, Errors: 0
utest/client/test_extrameta.c:97:F:Core:test_extrametas:0: Assertion 'has_extrameta(path, cmd, 1, 1 )==0' failed
utest/client/protocol2/test_rabin_read.c:79:F:Core:test_rabin_read:0: Assertion 'has_xattr(sb->path.buf)==0' failed
utest/client/test_xattr.c:148:F:Core:test_xattrs:0: Assertion 'has_xattr(path)==0' failed
FAIL runner (exit status: 1)
```

**Comment by grke** [comment]
I will try again with everything fresh.

**Comment by grke** [comment]
My build got past the tests, but errored a bit later:
```
# remove duplicate files
rm debian/burp/usr/share/doc/burp/LICENSE
rm: cannot remove 'debian/burp/usr/share/doc/burp/LICENSE': No such file or directory
debian/rules:21: recipe for target 'override_dh_auto_install' failed
make[1]: *** [override_dh_auto_install] Error 1
make[1]: Leaving directory '/src'
debian/rules:7: recipe for target 'binary' failed
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
debuild: fatal error at line 1152:
dpkg-buildpackage -rfakeroot -us -uc -ui -i -b failed
FAIL
The command '/bin/sh -c REV=2;cd src && export MAKEOPTS="-j4" &&     if debuild -i -us -uc -b;then echo SUCCESS; else echo FAIL; exit 1;fi' returned a non-zero code: 1
graham@small:/tmp/burp$ 
```

Anyway, by following your procedure, I cannot reproduce your issue.
It is my belief that, somehow, xattr support is not enabled for you inside your container, but are for me.

Is there anything that you would like me to check?

Did you check "mount|grep xattr" inside your container?
What was the output?

**Comment by kiorky** [comment]
I confirm that xattr wasnt enabled, i remember that the remount call did not worked as anted.

I think that the tests with xattr must be skipped if the underlying FS does not support it.

**Comment by grke** [comment]
Hello,

If you skip the xattr tests, how can you be sure that the binary that you built works properly on systems that have xattrs?

If you don't want xattr supprt, shouldn't you just run 'configure' with '--enable-xattr=no'?

('--enable-xattr=no' appears to be broken at the moment, I will fix that for the next release)

**Comment by kiorky** [comment]
Because you often run builds for binairies under platforms that wont support it (like Travis, launchpad, gitlabci, or other CI farms,  or when deploying onto kubernetes/docker based) and then you want to run 80% of the test suite assuming that the other tests that need special features/support had been done earlier in the deploy pipeline ensuring the global consistency.
As those tests are integrated automaticly by default in the packaging glues (like debian-helper, or portage, etc), it's a best practise to have a condtional skip on tests that need sepcial support.

**Comment by kiorky** [comment]
So yes i agree that you as the maintainer and author must ensure that xattr support is well tested and not skipped on your (MAIN) test process, but it would be super cool to have a conditional to help regular packaging :).

**Comment by grke** [comment]
This was fixed in burp 2.2.14 "Don't run xattr tests if the filesystem doesn't support them."
Change-Id: I368683b2850aeccf65f1919f6149f775b2b5ec3c
Thanks for the suggestions.
Closing.

**Comment by kiorky** [comment]
For the record, https://github.com/grke/burp/commit/2fca7bf91670204bf7f32d94c6450484ccf660dc

A big thx !
